ocaml/xapi/repository.ml

```diff
@@ -65,6 +65,24 @@ let introduce_bundle ~__context ~name_label ~name_description =
      ) ;
   create_repository_record ~__context ~name_label ~name_description
     ~binary_url:"" ~source_url:"" ~update:true ~gpgkey_path:"" ~origin:`bundle
+    ~address:"" ~certificate:""
+
+let introduce_remote_pool ~__context ~name_label ~name_description ~address
+    ~certificate =
+  Db.Repository.get_all ~__context
+  |> List.iter (fun ref ->
+         if
+           name_label = Db.Repository.get_name_label ~__context ~self:ref
+           || address = Db.Repository.get_address ~__context ~self:ref
+         then
+           raise
+             Api_errors.(
+               Server_error (repository_already_exists, [Ref.string_of ref])
```

Comment:
How about adding a new error `remote_pool_repository_already_exists` for the existing same address?

Reply:
I think we can reuse this error to simplify this as we will also raise the error when the name_label already exists.